app/models/chargeback/consumption_without_rollups.rb

```diff
@@ -33,20 +33,29 @@ def parents_determining_rate
        MiqEnterprise.my_enterprise].compact
     end
 
-    def none?(_metric)
-      true # No, values except for fixed (RateDetail.fixed?)
-    end
-
-    def max(_metric)
-      raise NotImplementedError # Unreachable code since none?==true
-    end
-
-    def avg(_metric)
-      raise NotImplementedError # Unreachable code since none?==true
+    def none?(metric)
+      current_value(metric).nil?
     end
 
     def chargeback_fields_present
       1 # Yes, charge this interval as fixed_compute_*_*
     end
+
+    def current_value(metric)
+      # Return the last seen allocation for charging purposes.
+      @value ||= {}
+      @value[metric] ||= case metric
+                         when 'derived_vm_numvcpus' # Allocated CPU count
+                           resource.hardware.try(:cpu_total_cores)
+                         when 'derived_memory_available'
+                           resource.hardware.try(:memory_mb)
+                         when 'derived_vm_allocated_disk_storage'
+                           resource.allocated_disk_storage
+                         end
+      @value[metric]
+    end
+    alias avg current_value
```

Comment:
thanks.

my broader context:

Lets get the relationship of these columns into the model itself. We have `has_many` and for aggregation I've been using `virtual_aggregate`. This paint a much bigger picture than a simple ruby method or a `virtual_attribute`. Not sure if a `virtual_aggregate` works for you, or if you need a different mechanism.

In the short term, aggregation may be represented as an n+1 query or a subquery in the main query.
But in the longer term, reports can split many aggregates into a separate query using `GROUP BY` or similar. The report will then merge these 2 result sets together.

But this can only be done when the framework can discern the relationship between models.

You have been doing a great job pulling these concepts out. And we both don't want to prematurely optimize. But it would be nice if we could use a declarative syntax so the relationships could be used to optimize our access to this data